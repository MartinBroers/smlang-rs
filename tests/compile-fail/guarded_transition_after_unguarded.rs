extern crate smlang;

use smlang::statemachine;

statemachine! {
    transitions: {
        State1 + Event1 = Fault,
        *State1 + Event1 [guard] = State2,
    }
}

fn main() {}
