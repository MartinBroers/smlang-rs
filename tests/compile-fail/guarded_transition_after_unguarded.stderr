error: State1 + Event1: [guard()] : guarded transition is unreachable because it follows an unguarded transition, which handles all cases
  --> tests/compile-fail/guarded_transition_after_unguarded.rs:5:1
   |
5  | / statemachine! {
6  | |     transitions: {
7  | |         State1 + Event1 = Fault,
8  | |         *State1 + Event1 [guard] = State2,
9  | |     }
10 | | }
   | |_^
   |
   = note: this error originates in the macro `statemachine` (in Nightly builds, run with -Z macro-backtrace for more info)
